net/strongswan/files/ipsec.init

```diff
@@ -58,7 +60,24 @@ warning() {
 	echo "WARNING: $@" >&2
 }
 
-add_crypto_proposal() {
+fatal() {
+	echo "ERROR: $@" >&2
+
+	CONFIG_FAIL=1
+}
+
+is_aead() {
+	local cipher="$1"
+
+	case "$cipher" in
```

Comment:
You could just generalize that to *gcm*|*ccm*|*gmac*, because those are all AEAD cipher modes. Meaning, if potentially, some algorithm xez was standardized, then xez*cbc, xez*gcm*, xez*ccm*, xez*gmac* could occur and the current expression would not catch these parts of a proposal.

Reply:
Understood, but this is the check we already have in `swanctl.init`

Reply:
Will handle `chacha20poly1305` as a follow-up PR.